grails-app/domain/com/coincollector/payment/Payment.groovy

```diff
@@ -31,6 +31,8 @@ class Payment extends BaseEntity {
         query { Map search ->
             if (search.containsKey("id")) eq("id", Long.valueOf(search.id))
 
+            if (search.containsKey("status")) eq("status", search.status)
```

Comment:
Quando temos named query com Enums, o ideal é sempre converte-la aqui também. É uma boa prática para evitar erros de desenvolvimento, pois caso o desenvolvedor esqueça de converter antes, aqui garante que o formato está correto.

```suggestion
            if (search.containsKey("status")) eq("status", PaymentStatus.valueOf(search.status.toString()))
```

Aqui é usado `Enum.valueOF` pois caso for um valor inválido, irá dar uma exception, ao contrário de usar convert que gera null.